pkg/clustermesh/mcsapi/serviceimport_controller_test.go

```diff
@@ -0,0 +1,463 @@
+		require.Nil(t, meta.FindStatusCondition(svcExport.Status.Conditions, mcsapiv1alpha1.ServiceExportConflict))
+	})
+
+	conflictTests := []struct {
+		name                 string
+		remoteSvcImportValid func(*mcsapiv1alpha1.ServiceImport) bool
+		localSvcImportValid  func(*mcsapiv1alpha1.ServiceImport) bool
+	}{
+		{
+			name: "conflict-type",
+			remoteSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.Type == mcsapiv1alpha1.Headless
+			},
+			localSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.Type == mcsapiv1alpha1.ClusterSetIP
+			},
+		},
+		{
+			name: "conflict-port-name",
+			remoteSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.Ports[0].Name == "remote"
+			},
+			localSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.Ports[0].Name == ""
+			},
+		},
+		{
+			name: "conflict-session-affinity",
+			remoteSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.SessionAffinity == corev1.ServiceAffinityClientIP
+			},
+			localSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return svcImport.Spec.SessionAffinity == corev1.ServiceAffinityNone
+			},
+		},
+		{
+			name: "conflict-session-affinity-config",
+			remoteSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return *svcImport.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == 42
+			},
+			localSvcImportValid: func(svcImport *mcsapiv1alpha1.ServiceImport) bool {
+				return *svcImport.Spec.SessionAffinityConfig.ClientIP.TimeoutSeconds == 4242
+			},
+		},
+	}
+
+	for _, conflictTest := range conflictTests {
+		t.Run("Conflict test "+conflictTest.name+" with remote conflict", func(t *testing.T) {
+			key := types.NamespacedName{
+				Name:      conflictTest.name,
+				Namespace: "default",
+			}
+			result, err := r.Reconcile(context.Background(), ctrl.Request{
+				NamespacedName: key,
+			})
+
+			require.NoError(t, err)
+			require.Equal(t, ctrl.Result{}, result, "Result should be empty")
+
+			svcImport, err := getServiceImport(c, key)
+			require.NoError(t, err)
+			require.NotNil(t, svcImport)
+			require.True(t, conflictTest.remoteSvcImportValid(svcImport))
+
+			svcExport, err := getServiceExport(c, key)
+			require.NoError(t, err)
+			require.NotNil(t, svcExport)
+			require.True(t, meta.IsStatusConditionFalse(svcExport.Status.Conditions, conditionTypeReady))
+			require.True(t, meta.IsStatusConditionTrue(svcExport.Status.Conditions, mcsapiv1alpha1.ServiceExportValid))
+			require.True(t, meta.IsStatusConditionTrue(svcExport.Status.Conditions, mcsapiv1alpha1.ServiceExportConflict))
+		})
+	}
+
+	for _, remoteFixture := range remoteSvcImportTestFixtures {
+		remoteFixture.ExportCreationTimestamp = newerTime
+	}
```

Comment:
This snipped modifies global variables, and apparently breaks the test when run with count > 1:

```
$ go test ./pkg/clustermesh/mcsapi/ --run Test_mcsServiceImport_Reconcile --count 2 

--- FAIL: Test_mcsServiceImport_Reconcile (0.00s)
    --- FAIL: Test_mcsServiceImport_Reconcile/Conflict_test_conflict-type_with_remote_conflict (0.00s)
        serviceimport_controller_test.go:422: 
                Error Trace:    /home/marco/Documents/isovalent/cilium/pkg/clustermesh/mcsapi/serviceimport_controller_test.go:422
                Error:          Should be true
                Test:           Test_mcsServiceImport_Reconcile/Conflict_test_conflict-type_with_remote_conflict
...
```